osu.Game/Overlays/BeatmapSet/BeatmapNotAvailable.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class BeatmapNotAvailable : Container
+    {
+        private BeatmapSetInfo beatmapSet;
+
+        public BeatmapSetInfo BeatmapSet
+        {
+            get => beatmapSet;
+            set
+            {
+                if (value == beatmapSet)
+                    return;
+
+                beatmapSet = value;
+
+                var unavailable = availability != null;
```

Comment:
should be `bool` not `var`. also this is wrong. what about if the object is present but has an empty URL and disabled=false?

Reply:
unavailable will return true, and it'll go through update text and pick parts-removed text, and will not add a link, i forgot to add a link.Clear while i was experimenting..